### PR TITLE
Change API to be more clear about exceptions

### PR DESCRIPTION
**Pull request by mikonieminen**

### CODE REVIEW

File: lib/letters.ml [hunk 1]
```diff
@@ -140,69 +140,72 @@ let build_email ~from ~recipients ~subject ~body =
...
+      let* res =
+        Sendmail_handler.run ~hostname ~port ~domain ~authentication
+          ~tls_authenticator ~from:from_addr ~recipients ~mail
+      in
+      match res with
+      | Ok () -> Lwt.return (Ok ())
+      | Error err ->
         Lwt.fail_with (Fmt.str "Sending email failed, %a" Sendmail.pp_error err)
+  with
+  | _ -> Lwt.fail_with "Unexpected internal failure"
```
Comment (joseferben) [inline lib/letters.ml]: This fails with an exception, if you want to return a `Result.t` you should use `Lwt_result.fail`

**Review by joseferben — CHANGES_REQUESTED** [review]
I start to think that throwing exceptions (that are truly exceptions) is ok. Let's talk about error handling in general again. But a function that returns a `Result.t` should never throw imo.

File: lib/letters.ml (hunk 1 above)
Comment (mikonieminen) [reply lib/letters.ml]: OK, I have to agree that my change was pretty badly prepared. How about now?

**Comment by mikonieminen** [comment]
After discussions and changes, the nature of this branch has changed and I want to rename the branch.

**Pull request closed without merging**